api/secrets.proto

```diff
@@ -0,0 +1,136 @@
+	// name prefix in `ListSecretsRequest.NamePrefix`
+	// - Returns an error if listing fails.
+	rpc ListSecrets(ListSecretsRequest) returns (ListSecretsResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+	// CreateSecret creates and return a `CreateSecretResponse` with a `Secret`` based
+	// on the provided `CreateSecretRequest.SecretDataSpec`.
+	// - Returns `InvalidArgument` if the `CreateSecretRequest.SecretDataSpec` is malformed.
+	// - Returns `AlreadyExists` if the `CreateSecretRequest.SecretDataSpec.Annotation.Name`
+	//   conflicts with an existing secret's name.
+	// - Returns an error if the creation fails.
+	rpc CreateSecret(CreateSecretRequest) returns (CreateSecretResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+
+	// UpdateSecret adds the `UpdateSecretRequest.SecretDataSpec` to a `Secret`` as a new version,
+	// (and is set as the the `Secret.LatestVersion`) and returns the updated
+	// `Secret` in the `UpdateSecretResponse`.
+	// - Returns `NotFound` if no secret with name
+	//   `UpdateSecretRequest.SecretDataSpec.Annotations.Name` is found.
+	// - Returns `InvalidArgument` if the `UpdateSecretRequest.SecretDataSpec` is malformed.
+	// - Returns an error if the update fails.
+	rpc UpdateSecret(UpdateSecretRequest) returns (UpdateSecretResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+
+	// RemoveSecret removes a referenced by `RemoveSecretRequest.Name`.  If
+	// `RemoveSecretRequest.Version` is not provided, the entire secret and all its
+	// versions are removed.  Otherwise, only the version of the secret referenced
+	// by `RemoveSecretRequest.Version` is removed from the secret.  The
+	// `Secret.LatestVersion` is updated as needed.
+	// - Returns `InvalidArgument` if `RemoveSecretRequest.Name` is empty.
+	// - Returns `NotFound` if the a secret named `RemoveSecretRequest.Name` is not
+	//   found or if it is found but has no version `RemoveSecretRequest.Version`.
+	// - Returns an error if the deletion fails.
+	rpc RemoveSecret(RemoveSecretRequest) returns (RemoveSecretResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+}
+
+// GetSecretRequest is the request to get a `Secret` object given a secret name.
+message GetSecretRequest {
+	string name = 1;
+}
+
+// GetSecretResponse contains the Secret corresponding to the name in
+// `GetSecretRequest`, but the `SecretData.Spec.Data` field in each `SecretData`
+// object in the `Secret` should be nil instead of actually containing
+// the secret bytes.
+message GetSecretResponse {
+	Secret secret = 1;
+}
+
+// ListSecretRequest is the request to list all secrets in the secret store,
+// or all secrets filtered by name or name prefix.
+message ListSecretsRequest {
+	message Filters {
+		repeated string names = 1;
+		repeated string name_prefixes = 2;
```

Comment:
@diogomonica Wrong.

It allows you to implement short-code lookup, so the user can do `abc` instead of `abcdef`.